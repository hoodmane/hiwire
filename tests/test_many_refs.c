#include "testlib.h"

int
main()
{
  int offset = 80000;
  HwRef refs[5000];
  for (int i = 0; i < 5000; i++) {
    refs[i] = hiwire_new(int_to_ref(offset + i));
  }
  for (int i = 0; i < 5000; i++) {
    if (ref_to_int(hiwire_get(refs[i])) != offset + i) {
      printf(FAIL_STR "\n");
      return -1;
    }
  }
  printf(PASS_STR "\n");
  for (int i = 0; i < 5000; i++) {
    hiwire_decref(refs[i]);
  }
  offset *= 6;
  for (int i = 0; i < 5000; i++) {
    refs[i] = hiwire_new(int_to_ref(offset + 2 * i));
  }
  for (int i = 0; i < 5000; i++) {
    if (ref_to_int(hiwire_get(refs[i])) != offset + 2 * i) {
      printf(FAIL_STR "\n");
      return -1;
    }
  }
  printf(PASS_STR "\n");
  return 0;
}
