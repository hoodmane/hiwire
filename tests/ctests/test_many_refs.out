[94;1mPASS [0m
[94;1mPASS [0m
